src/matrixrtc/EncryptionManager.ts

```diff
@@ -0,0 +1,500 @@
+import { type MatrixClient } from "../client.ts";
+import { logger as rootLogger } from "../logger.ts";
+import { MatrixEvent } from "../models/event.ts";
+import { Room } from "../models/room.ts";
+import { EncryptionConfig } from "./MatrixRTCSession.ts";
+import { secureRandomBase64Url } from "../randomstring.ts";
+import { EncryptionKeysEventContent } from "./types.ts";
+import { decodeBase64, encodeUnpaddedBase64 } from "../base64.ts";
+import { MatrixError, safeGetRetryAfterMs } from "../http-api/errors.ts";
+import { CallMembership } from "./CallMembership.ts";
+import { EventType } from "../@types/event.ts";
+const logger = rootLogger.getChild("MatrixRTCSession");
+
+/**
+ * A type collecting call encryption statistics for a session.
+ */
+export type Statistics = {
+    counters: {
+        roomEventEncryptionKeysSent: number;
+        roomEventEncryptionKeysReceived: number;
+    };
+    totals: {
+        roomEventEncryptionKeysReceivedTotalAge: number;
+    };
+};
+
+/**
+ * This interface is for testing and for making it possible to interchange the encryption manager.
+ * @internal
+ */
+export interface IEncryptionManager {
+    join(joinConfig: EncryptionConfig | undefined): void;
+    leave(): void;
+    onMembershipsUpdate(oldMemberships: CallMembership[]): Promise<void>;
+    onCallEncryptionEventReceived(event: MatrixEvent): void;
```

Comment:
```suggestion
    /**
     * Process `m.call.encryption_keys` events to track the encryption keys for call participants.
     * This should be called each time the relevant event is received from a room timeline.
     * If the event is malformed then it will be logged and ignored.
     *
     * @param event the event to process
     */
    onCallEncryptionEventReceived(event: MatrixEvent): void;
```